mlrun/launcher/remote.py

```diff
@@ -90,9 +90,11 @@ def launch(
                 runtime.deploy(skip_deployed=True, show_on_failure=True)
 
             else:
-                raise mlrun.errors.MLRunRuntimeError(
-                    "Function image is not built/ready, set auto_build=True or use .deploy() method first"
-                )
+                if not runtime.spec.build.load_source_on_run:
```

Comment:
I think that checking `load_source_on_run` is not enough. You need to do something like
~~~
require_build = (
            build.commands
            or build.requirements
            or (build.source and not build.load_source_on_run)
        )
~~~
like in `prepare_image_for_deploy`. Maybe extract it to a method under BaseRuntime